prebid_cache_client/client.go

```diff
@@ -47,13 +50,21 @@ func NewClient(conf *config.Cache) Client {
 				IdleConnTimeout: 65,
 			},
 		},
-		putUrl: conf.GetBaseURL() + "/cache",
+		putUrl:  conf.GetBaseURL(true, false),
+		urlHost: conf.GetHost(),
+		urlPath: conf.GetPath(),
```

Comment:
Why do we need `urlHost` & `urlPath` here when we have the `putUrl` which is basically made of those values? I thought we agreed on just having the `putUrl` field here and changing the `GetBaseURL` method to return the constructed URL string (should probably change the name of that new method to `GetURL`). And then for the targeting data, you can alter the `GetPrebidCacheSplitURL` method to parse the string into a `url.URL` object and return the `host` and the `path` values. 

Or an even better approach would be to just change the `putUrl` type to be `url.URL` so that you don't have to switch between a `ur.URL` object and it's string representation back and forth